File: PaymentManagement/src/main/java/com/paymentservice/paymentmanagement/configurations/Swagger.java
package com.paymentservice.paymentmanagement.configurations;

import io.swagger.v3.oas.annotations.OpenAPIDefinition;
import io.swagger.v3.oas.annotations.info.Contact;
import io.swagger.v3.oas.annotations.info.Info;
import io.swagger.v3.oas.annotations.servers.Server;

@OpenAPIDefinition(
    info = @Info(
            contact = @Contact(
                    name = "Dave CHEDJOUN",
                    email = "davechedjoun@gmail.com"
    ),
            title = " Documentation for Payment management service",
            description = "All routes to resources of the api with their description",
            version = "1.0"
),
        servers = {
            @Server(
                    description = "local env",
                    url = "http://localhost:8083/"
            ),
            @Server(
                    description = "prod env",
                    url = ""
            )
        }
)
public class Swagger {


}

File: InventoryManagament/swagger.js
const swaggerUi = require('swagger-ui-express');
const swaggerJsdoc = require('swagger-jsdoc');

const options = {
  definition: {
    openapi: '3.0.0',
    info: {
      title: 'Inventory Management API',
      version: '1.0.0',
      description: "A simple express API for Inventory Management Services"
    },

    servers: [
      {
        url: "http://localhost:3002",
        description: "Local server"
      },
      {
        url: "https://propelize.com",
        description: "Production server"
      },
    ]
  },
  apis: ["./models/*", "./routes/*"], // files containing annotations as above
}

const specs = swaggerJsdoc(options);

module.exports = (app) => {
  app.use("/api-docs", swaggerUi.serve, swaggerUi.setup(specs))
}